client/errors.go

```diff
@@ -0,0 +1,13 @@
+package client
+
+import "fmt"
+
+type ResponseError struct {
+	URL        string
+	StatusCode int
+	Body       []byte
+}
+
+func (r ResponseError) Error() string {
+	return fmt.Sprintf("error calling %s, got status code %d", r.URL, r.StatusCode)
```

Comment:
Can we spit out the `Body []byte` in the `Error()` message? I've found that it helps when debugging an error.

Reply:
Makes sense to me